lib/grpnames.gi

```diff
@@ -20,46 +20,575 @@
 ##  from GAP3 code written by Markus Püschel and Sebastian Egner.
 ##
 
+
+#############################################################################
+##
+#M  IsTrivialNormalIntersectionInList( <L>, <U>, <V> ) . . . . generic method
+##
+InstallMethod( IsTrivialNormalIntersectionInList,
+               "if minimal normal subgroups are computed",
+               [ IsList, IsGroup, IsGroup ],
+
+  function( L, U, V )
+    local gs;
+
+    gs:= List(Filtered(L, N -> not IsTrivial(N)),
+                                               N -> GeneratorsOfGroup(N)[1]);
+    return ForAll(gs, g -> not g in U or not g in V);
+  end);
+
+#############################################################################
+##
+#M  IsTrivialNormalIntersection( <G>, <U>, <V> ) . . . . . . . generic method
+##
+
+InstallMethod( IsTrivialNormalIntersection,
+               "if minimal normal subgroups are computed",
+               [ IsGroup and HasMinimalNormalSubgroups, IsGroup, IsGroup ],
+
+  function( G, U, V )
+    local gs;
+
+    gs := List(MinimalNormalSubgroups(G), N -> GeneratorsOfGroup(N)[1]);
+    return ForAll(gs, g -> not g in U or not g in V);
+  end);
+
+InstallMethod( IsTrivialNormalIntersection,
+               "if socle is computed",
+               [ IsGroup and HasSocle, IsGroup, IsGroup ],
+
+  function( G, U, V )
+    local gs;
+
+    gs := GeneratorsOfGroup(Socle(G));
+    return ForAll(gs, g -> not g in U or not g in V);
+  end);
+
+InstallMethod( IsTrivialNormalIntersection,
+               "for nilpotent groups",
+               [ IsGroup and IsNilpotentGroup, IsGroup, IsGroup ],
+
+  function( G, U, V )
+    local gs;
+
+    gs := List(MinimalNormalSubgroups(Center(G)),
+                                              N -> GeneratorsOfGroup(N)[1]);
+    return ForAll(gs, g -> not g in U or not g in V);
+  end);
+
+InstallMethod( IsTrivialNormalIntersection,
+               "generic method", [ IsGroup, IsGroup, IsGroup ],
+
+  function( G, U, V )
+
+    return IsTrivial(NormalIntersection(U, V));
+  end);
+
+#############################################################################
+##
+#M  ComplementNormalSubgroup( <G>, <N> ) . . . . . . . . . . . generic method
+##
+InstallMethod( ComplementNormalSubgroup,
+               "generic method", [ IsGroup,  IsGroup ],
+
+  function( G, N )
+
```

Comment:
Since `N` is supposed to be a subgroup of `G`, the family predicate of this method should reflect that -- i.e. you should insert `IsIdenticalObj` as argument  between the function name and the doc strings.